game-core/src/main/java/games/strategy/engine/framework/GameRunner.java

```diff
@@ -81,30 +81,27 @@ private GameRunner() {}
    */
   public static void start() {
     SwingUtilities.invokeLater(() -> {
-      final JFrame frame = JFrameBuilder.builder()
-          .title("TripleA")
-          .build();
-      setupPanelModel = new SetupPanelModel(gameSelectorModel, frame);
-      mainFrame = newMainFrame(frame);
+      newMainFrame();
```

Comment:
Could this method still return the frame that was created?

I note this as the method currently has all side effects. The initialization of `mainFrame` is not localized anymore and I think that will make it harder for a future maintainer.

Also, with a return value, proper call ordering would be easier to understand. It's generally a bad sign when code can be re-ordered, still compile, and then be wrong.

Reply:
Yeah, the side effects are weird, however I didn't want to make the mainFrame public just to be able to set it in the process. I removed the return type on purpose, so that it was clear that there are side-effects.
If you have a proposal on how to change the methods, so the mainFrame can still be set via a public method and your preferences, I'd be happy to use it.